cachito/web/api_v1.py

```diff
@@ -29,12 +33,27 @@ def get_requests():
         value exceeds configuration's CACHITO_MAX_PER_PAGE.
     :rtype: flask.Response
     """
-    # Default versbose flag to False
+    # Check if user is filtering requests by state
+    state = flask.request.args.get('state')
+    # Default verbose flag to False
     verbose = str_to_bool(flask.request.args.get('verbose', False))
     max_per_page = flask.current_app.config['CACHITO_MAX_PER_PAGE']
     # The call to `paginate` will inspect the current HTTP request for the
     # pagination parameters `page` and `per_page`.
-    pagination_query = Request.query.paginate(max_per_page=max_per_page)
+    query = Request.query
+    if state:
+        if state not in ['in_progress', 'complete', 'failed', 'stale']:
```

Comment:
If we ever add a state, we'll need to update this list. My suggestion is to use `cachito.web.models.RequestStateMapping.get_state_names()`.

An alternative is to do:
```python
try:
    state_int = RequestStateMapping.__members__[state].value
except KeyError:
    raise ValidationError('error goes here')
```